src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -235,10 +240,26 @@ internal static WorkerInitResponse WorkerInitRequestHandler(WorkerInitRequest re
             response.Capabilities.Add("RpcHttpTriggerMetadataRemoved", bool.TrueString);
             response.Capabilities.Add("UseNullableValueDictionaryForHttp", bool.TrueString);
             response.Capabilities.Add("TypedDataCollection", bool.TrueString);
+            response.Capabilities.Add("HandlesWorkerTerminateMessage", bool.TrueString);
 
             return response;
         }
 
+        internal static void WorkerTerminateRequestHandler(WorkerTerminate request)
+        {
+            // Terminate the worker process
+            try
+            {
+                var workerInformation = WorkerInformation.Instance;
+                var workerProcess = Process.GetProcessById(workerInformation.ProcessId);
+                workerProcess.Kill();
```

Comment:
Don't we also need to expose a Cancellation token instance to user's function code so that they can check the `IsCancellationRequested` property and act accordingly. May be it can be a property on the `FunctionContext` or a method parameter they can inject to the function method.